index.js

```diff
@@ -22,83 +22,407 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
-import sh from 'execSync';
+//import sh from 'execSync';
+import i2c from 'i2c-bus';
+import { execSync } from 'child_process';
 import { Peripheral } from 'raspi-peripheral';
+import { VERSION_1_MODEL_B_REV_1, getBoardRevision } from 'raspi-board';
+
+if (typeof execSync !== 'function') {
+  execSync = require('execsync');
+}
+
+function checkAlive(alive) {
+  if (!alive) {
+    throw new Error('Attempted to access a destroyed I2C peripheral');
+  }
+}
+
+function checkAddress(address) {
```

Comment:
Looking at the code more, I think that the reset of the helper methods make sense as stand-alone functions. ```checkAlive``` is the only one that really needs to be moved into the class.